Sources/OTel/Helpers/Timeout.swift

```diff
@@ -0,0 +1,38 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2024 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+package func withTimeout<ClockType: Clock, ChildTaskResult>(
```

Comment:
Nice addition! Looking forward to reusing this in `OTelBatchSpanProcessor` where possible.